### PR TITLE
feat: limit the name given to `ImportSpecifier` in `id-length`

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
Right now `id-length` rule is not reporting the name specified to import specifier, made the rule to limit the name given to `ImportSpecifier`.

```js
/*eslint id-length: ["error", { min: 4 }]*/

import { x as foo } from 'module'; // foo will be reported.
```

```js
/*eslint id-length: ["error", { max: 5 }]*/

import { x as myVariable } from 'module'; // myVariable will be reported.
```

### CODE REVIEW

**Comment by mdjermanovic** [comment]
> ```js
> /*eslint id-length: ["error", { min: 4 }]*/
> 
> import { x as foo } from 'module'; // foo will be reported.
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> ```js
> /*eslint id-length: ["error", { max: 5 }]*/
> 
> import { x as myVariable } from 'module'; // myVariable will be reported.
> ```

To clarify, the rule would check only renamed imports?

**Comment by Tanujkanti4441** [comment]
> To clarify, the rule would check only renamed imports?
> 

Yes, only renamed imports or local name assigned to imports

**Comment by nzakas** [comment]
I'm not sure I understand what's going on here. Is this a bug? If so, it would be helpful to see a before and after example.

(`id-length` is a stylistic rule, so it's frozen except for bug fixes.)

**Comment by Tanujkanti4441** [comment]
> I'm not sure I understand what's going on here. Is this a bug? If so, it would be helpful to see a before and after example.
> 


sure, right now `id-length` rule doesn't report the local name assigned to `importSpecifier`

**Before**
```js
/*eslint id-length: ["error", { min: 4 }]*/

import { x as foo } from 'module';  // rule doesn't report 'foo' even it's length is short (less then 4).
```
**After**

```js
/*eslint id-length: ["error", { min: 4 }]*/

import { x as foo } from 'module'; // rule will report 'foo'.

import { x } from 'module'; // but rule will not report 'x'.
```

> (`id-length` is a stylistic rule, so it's frozen except for bug fixes.)

it is similar to issue #18847, so i think it can also fall under bug fix.

File: lib/rules/id-length.js
Comment (mdjermanovic) [inline lib/rules/id-length.js:121]: I think we should compare by names, so that `import { x as x }` is not reported because it's basically the same as `import { x }`.

Comment (mdjermanovic) [reply lib/rules/id-length.js:121]: We could use [`astUtils.getModuleExportName`](https://github.com/eslint/eslint/blob/59cfc0f1b3bbb62260602579f79bd1c36ab5a00f/lib/rules/utils/ast-utils.js#L792) for `parent.imported` so that `import { "x" as x }` also isn't reported.

Comment (Tanujkanti4441) [reply lib/rules/id-length.js]: Done!

File: tests/lib/rules/id-length.js
Comment (mdjermanovic) [inline tests/lib/rules/id-length.js:69]:
```suggestion
        { code: "import { x as x } from 'y';", languageOptions: { ecmaVersion: 6, sourceType: "module" } },
        { code: "import { 'x' as x } from 'y';", languageOptions: { ecmaVersion: 2022, sourceType: "module" } },
```

It would be good to add a test with a string literal import name.

File: tests/lib/rules/id-length.js
Comment (mdjermanovic) [inline tests/lib/rules/id-length.js:256]:
```suggestion
        { code: "import { x as z } from 'module';", languageOptions: { ecmaVersion: 6 }, errors: [{ ...tooShortError, column: 15 }] },
        { code: "import { foo as z } from 'module';", languageOptions: { ecmaVersion: 6 }, errors: [{ ...tooShortError, column: 17 }] },
        { code: "import { 'foo' as z } from 'module';", languageOptions: { ecmaVersion: 2022 }, errors: [{ ...tooShortError, column: 19 }] },
```

It would be good to check the `column` to verify that the right identifier is reported, and also add a test where the imported name is of a valid length and a test with string literal.

File: tests/lib/rules/id-length.js
Comment (mdjermanovic) [inline tests/lib/rules/id-length.js:274]:
```suggestion
            errors: [{ ...tooLongError, column: 17 }]
```

**Review by mdjermanovic — COMMENTED** [review]
Code LGTM, I only have a few suggestions about the tests.

File: tests/lib/rules/id-length.js
Comment (Tanujkanti4441) [reply tests/lib/rules/id-length.js]: Added!

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Since this is a `feat`, it needs a second review before merging.

**Pull request merged**
